packages/toolpad-app/pages/deploy/[appId]/[[...path]].tsx

```diff
@@ -26,7 +28,7 @@ export const getServerSideProps: GetServerSideProps<ToolpadAppProps> = async (co
   //   context.res.setHeader('X-Frame-Options', 'DENY');
   // }
 
-  if (!app.public) {
+  if (!app.public && !config.isDemo) {
```

Comment:
```suggestion
  if (!app.public) {
```

`checkBasicAuth` below should return `true` if no basic auth is configured. The demo doesn't configure basic auth so this extra condition shouldn't be necessary